### ISSUE TITLE
Functional enum yields "Illegal type annotation: variable not allowed unless it is a type alias"

### ISSUE THREAD

**Issue by schra** [microsoft/pyright#4363]
**What**: Using the functional syntax of `enum.Enum`, pyright throws the error `error: Illegal type annotation: variable not allowed unless it is a type alias`.

**Steps**: Create the file `foo.py`:

```
#!/usr/bin/env python3

from enum import Enum

Foo = Enum('Foo', ['bar', 'baz'])

def identity(baz: Foo):
    return baz

print(identity(Foo.bar))
```

Then call `pyright` on that file:

```
$ python --version
Python 3.10.8

$ pyright --version
pyright 1.1.284

$ pyright foo.py
No configuration file found.
No pyproject.toml file found.
stubPath /tmp/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
pyright 1.1.284
/tmp/foo.py
  /tmp/foo.py:7:19 - error: Illegal type annotation: variable not allowed unless it is a type alias (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
Completed in 0.802sec

$ mypy --version
mypy 0.991 (compiled: no)

$ mypy foo.py
Success: no issues found in 1 source file
```

**Comment by erictraut** [comment]
Yes, this is intended behavior. You should use the class form of `Enum` if you want to use your enumeration with a static type checker. The only reason to use the functional form is to define a dynamic enumeration, and those won't work with static type checking anyway.
